File: mega-zep-frontend/src/main/angular/frontend/src/app/modules/employees/models/Employee.ts
export class Employee {
  private _userId: string;
  private _sureName: string;
  private _firstName: string;
  private _eMail: string;
  private _role: number;
  private _title: string;
  private _releaseDate: string;
  private _workDescription: string;
  private _salutation: string;

  get userId(): string {
    return this._userId;
  }

  set userId(value: string) {
    this._userId = value;
  }

  get sureName(): string {
    return this._sureName;
  }

  set sureName(value: string) {
    this._sureName = value;
  }

  get firstName(): string {
    return this._firstName;
  }

  set firstName(value: string) {
    this._firstName = value;
  }

  get eMail(): string {
    return this._eMail;
  }

  set eMail(value: string) {
    this._eMail = value;
  }

  get salutation(): string {
    return this._salutation;
  }

  set salutation(value: string) {
    this._salutation = value;
  }

  get workDescription(): string {
    return this._workDescription;
  }

  set workDescription(value: string) {
    this._workDescription = value;
  }

  get releaseDate(): string {
    return this._releaseDate;
  }

  set releaseDate(value: string) {
    this._releaseDate = value;
  }

  get title(): string {
    return this._title;
  }

  set title(value: string) {
    this._title = value;
  }

  get role(): number {
    return this._role;
  }

  set role(value: number) {
    this._role = value;
  }
}

File: mega-zep-backend/src/main/java/com/gepardec/mega/domain/model/employee/Employee.java
package com.gepardec.mega.domain.model.employee;

import com.google.auto.value.AutoValue;

import javax.annotation.Nullable;

@AutoValue
public abstract class Employee {

    @Nullable public abstract String userId();
    @Nullable public abstract String email();
    @Nullable public abstract String title();
    @Nullable public abstract String firstName();
    @Nullable public abstract String sureName();
    @Nullable public abstract String salutation();
    @Nullable public abstract String releaseDate();
    @Nullable public abstract String workDescription();
    @Nullable public abstract Integer role();
    public abstract boolean active();

    public static Builder builder() {
        return new com.gepardec.mega.domain.model.AutoValue_Employee.Builder()
                .active(false);
    }

    @AutoValue.Builder
    public abstract static class Builder {
        public abstract Builder userId(String userId);
        public abstract Builder email(String email);
        public abstract Builder title(String title);
        public abstract Builder firstName(String firstName);
        public abstract Builder sureName(String sureName);
        public abstract Builder salutation(String salutation);
        public abstract Builder releaseDate(String releaseDate);
        public abstract Builder workDescription(String workDescription);
        public abstract Builder role(Integer role);
        public abstract Builder active(boolean active);
        public abstract Employee build();
    }
}
